javaparser-core-testing/src/test/java/com/github/javaparser/CommentsInserterTest.java

```diff
@@ -112,4 +113,24 @@ void issue412() throws IOException {
         CompilationUnit cu = parseSample("Issue412").getResult().get();
         assertEqualToTextResourceNoEol(makeExpectedFilename("Issue412"), cu.toString());
     }
+
+    @Test
+    void lineCommentAssignedToLastNode() {
```

Comment:
Instead, the test should verify that the comment is backed to the correct node. Here the test is not sufficient because it tests that the pretty printer displays the comment on the last node. In addition, the test class is badly chosen to carry out this test which concerns the pretty printer.